### PR TITLE
Added flatpak build manifests with split dependencies

### PR DESCRIPTION
**Pull request by kdcis**
- Added Flatpak build manifests under ```packages/flatpak/``` directory

- Added Flatpak build instructions under ```docs/``` directory

- Added Dockerfile for Flatpak repo server  under ```packages/flatpak/``` directory

### CODE REVIEW

File: packages/flatpak/org.bitmessage.PyBitmessage.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        "build-options": {
+          "build-args": [
+            "--share=network"
+          ]
+        },
+        "build-commands": [
+          "python --version",
+          "pwd",
+          "ls",
+          "python checkdeps.py",
```
Comment (PeterSurda) [inline packages/flatpak/org.bitmessage.PyBitmessage.json:29]: lines 26 - 29 aren't necessary

File: packages/flatpak/org.bitmessage.PyBitmessage.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          "python setup.py install --prefix=/app --exec-prefix=/app",
+          "sed -i 's~/usr/bin/~/app/bin/~' /app/bin/pybitmessage",
+          "cat /app/bin/pybitmessage",
```
Comment (PeterSurda) [inline packages/flatpak/org.bitmessage.PyBitmessage.json:32]: this isn't necessary

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Please remove the extra lines from build sequence

**Comment by kdcis** [comment]
Removed extra lines from build sequence,
Updated Description,
Rebase done,
Added Build instructions under ```docs``` directory.

**Pull request closed without merging**